src/webserver.cpp

```diff
@@ -527,9 +567,14 @@ MHD_Result webserver::requests_answer_second_step(MHD_Connection* connection, co
 #ifdef DEBUG
         std::cout << "Writing content: " << std::string(upload_data, *upload_data_size) << std::endl;
 #endif  // DEBUG
-        mr->dhr->grow_content(upload_data, *upload_data_size);
+        if (mr->pp == nullptr || put_processed_data_to_content) {
```

Comment:
No, it's not redundant.

The post iterator is only created from the libmicrohttpd for content of type `multipart/form-data` and `application/x-www-form-urlencoded`. Otherwise the `MHD_create_post_processor` will not set up a post processor and return `nullptr`.

So even if the libhttpserver is used with `put_processed_data_to_content` to `false`, other content for which no post processor is created (i.e. `application/json`) must be put to the content, because the `get_content` of the libhttpserver is the only way to access this content.

So
- `mr->pp == nullptr` is the indication, that it is neither of the above two content types and therefor has to be put to content
- `put_processed_data_to_content` indicates, that data, which is passed to the post processor should additionally be put into the content (which is the default to preserve the current behavior)

Reply:
Given the question from @LeSpocky, let's add a comment clarifying what the if statement is doing.